cmd/algorand-indexer/daemon.go

```diff
@@ -65,276 +67,338 @@ var (
 	cpuProfile                string
 	pidFilePath               string
 	configFile                string
-)
+	suppliedAPIConfigFile     string
+	genesisJSONPath           string
+}
+
+var daemonCfg = &daemonConfig{}
 
 var daemonCmd = &cobra.Command{
 	Use:   "daemon",
 	Short: "run indexer daemon",
 	Long:  "run indexer daemon. Serve api on HTTP.",
+	//Args:
 	Run: func(cmd *cobra.Command, args []string) {
-		var err error
-		config.BindFlags(cmd)
-		err = configureLogger()
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "failed to configure logger: %v", err)
+		daemonCfg.flags = cmd.Flags()
+		if err := runDaemon(daemonCfg); err != nil {
 			panic(exit{1})
 		}
+	},
+}
 
-		if indexerDataDir == "" {
-			fmt.Fprint(os.Stderr, "indexer data directory was not provided")
-			panic(exit{1})
+func configureIndexerDataDir(cfg *daemonConfig) error {
+	var err error
+	if cfg.indexerDataDir == "" {
+		err = fmt.Errorf("indexer data directory was not provided")
+		logger.WithError(err).Errorf("indexer data directory error, %v", err)
+		return err
+	}
+	if _, err = os.Stat(cfg.indexerDataDir); os.IsNotExist(err) {
+		err = os.Mkdir(cfg.indexerDataDir, 0755)
+		if err != nil {
+			logger.WithError(err).Errorf("indexer data directory error, %v", err)
+			return err
 		}
-		// Create the directory if it doesn't exist
-		if _, err := os.Stat(indexerDataDir); os.IsNotExist(err) {
-			err := os.Mkdir(indexerDataDir, 0755)
-			maybeFail(err, "failure while creating data directory: %v", err)
+	}
+	return err
+}
+
+func loadIndexerConfig(cfg *daemonConfig) error {
+	var err error
+	indexerConfigFound := util.FileExists(filepath.Join(cfg.indexerDataDir, autoLoadIndexerConfigName))
+	if indexerConfigFound {
+		if cfg.configFile != "" {
+			err = fmt.Errorf("indexer configuration was found in data directory (%s) as well as supplied via command line.  Only provide one",
+				filepath.Join(cfg.indexerDataDir, autoLoadIndexerConfigName))
+			logger.WithError(err)
+			return err
+		}
+		// No config file supplied via command line, auto-load it
+		configs, err := os.Open(cfg.configFile)
+		if err != nil {
+			logger.WithError(err).Errorf("%v", err)
+			return err
+		}
+		defer configs.Close()
+		err = viper.ReadConfig(configs)
+		if err != nil {
+			logger.WithError(err).Errorf("invalid config file (%s): %v", viper.ConfigFileUsed(), err)
+			return err
+		}
```

Comment:
nit: We solved this slightly differently today on `develop` (also, I think this is opening the wrong file)
```suggestion
		// No config file supplied via command line, auto-load it
		configFile = filepath.Join(indexerDataDir, autoLoadIndexerConfigName)
	}
```

Reply:
I tried to match what was merged in from develop. In the most recent update. Let me know if the logic is not correct.

Reply:
The routine you added looks good.